### PR TITLE
release block earlier in SpillHandler

### PR DESCRIPTION
**Pull request by windtalker**
### What problem does this PR solve?

Issue Number: ref #6528

Problem Summary:

### What is changed and how it works?
release block once it is written to file in SpillHandler
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Core/SpillHandler.cpp [hunk 1]
```diff
@@ -73,7 +73,7 @@ bool SpillHandler::isSpilledFileFull(UInt64 spilled_rows, UInt64 spilled_bytes)
     return (spiller->config.max_spilled_rows_per_file > 0 && spilled_rows >= spiller->config.max_spilled_rows_per_file) || (spiller->config.max_spilled_bytes_per_file > 0 && spilled_bytes >= spiller->config.max_spilled_bytes_per_file);
 }
 
-void SpillHandler::spillBlocks(const Blocks & blocks)
+void SpillHandler::spillBlocks(Blocks & blocks)
```
Comment (SeaRise) [inline dbms/src/Core/SpillHandler.cpp:76]:
```suggestion
void SpillHandler::spillBlocks(Blocks && blocks)
```

**Review by SeaRise — APPROVED** [review]
Others LGTM

**Review by mengxin9014 — APPROVED** [review]
LGTM

**Pull request merged**